eng/helix/helix.proj

```diff
@@ -37,8 +40,15 @@
   <ItemGroup>
     <!--
       Use the BrowserDebugHost transport package as a sentinel for the non-shipping version of the NETCoreApp shared framework.
+      !temporary! e.g. https://dotnetbuilds.blob.core.windows.net/internal/Runtime/6.0.2-servicing.22060.12/dotnet-runtime-6.0.2-win-x86.zip?{token}
     -->
-    <AdditionalDotNetPackage Include="$(MicrosoftNETCoreBrowserDebugHostTransportVersion)">
+    <HelixCorrelationPayload Include="dotnet-cli"
+        Condition="'$(SYSTEM_TEAMPROJECT)' == 'internal'">
+        Destination="dotnet-cli"
+        Uri="https://dotnetbuilds.blob.core.windows.net/internal/Runtime/$(MicrosoftNETCoreBrowserDebugHostTransportVersion)/dotnet-runtime-$(MicrosoftNETCoreAppRuntimeVersion)-$(DotNetCliRuntime)$(ArchiveExtension)$([System.Environment]::GetEnvironmentVariable('DotNetBuildsInternalReadSasToken'))" />
+      
+    <AdditionalDotNetPackage Include="$(MicrosoftNETCoreBrowserDebugHostTransportVersion)"
+                             Condition="'$(SYSTEM_TEAMPROJECT)' == 'internal'">
       <PackageType>runtime</PackageType>
     </AdditionalDotNetPackage>
```

Comment:
We _always_ want to use the incoming .NET runtime when testing on Helix. Unless there's an alternative here, we're losing real coverage.

Reply:
I meant for this to be `!=` (i.e. use the `HelixCorrelationPayload` workaround in internal builds, stick with the old `AdditionalDotNetPackage` in public builds). But again, not sure that's right, as it's really only when the package we want comes from the internal runtime feed, which won't be the case in _every_ internal build

Reply:
Right. And, if you leave the checks here, will need both token properties set in the YAML.

One thought: The tokens are already available only in internal builds. When some special new variable is set in the AzDO UI, use the `/internal` URI and corresponding token. Keep the property names generic pass the right one from the YAML.

Then, change this stuff to choose based on the new variable too, not `$(SYSTEM_TEAMPROJECT)`.

Reply:
How would that work for PRs? We don't have control of queue time variables for PRs, right? Only for manually queued internal builds

Reply:
I was thinking of a property set at https://dev.azure.com/dnceng/internal/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=21&view=Tab_Variables and left alone except when we switch between the internal runtime feed and the regular one. Shouldn't show up in PRs except in the "this is defined and you can override it if you're doing the first PR for a switch on or off.

Reply:
I hesitate to add properties to the AzDO UI, that feels extremely "special" and prone to being ignored/forgotten/not known about - what if we just stuck with what we had pre-workaround, and made Helix ContinueOnError until https://github.com/dotnet/arcade/issues/8336 is done? Without it there's always some weird dance we'll have to do to determine exactly where to download runtime bits from, and we'll still get test coverage in public builds. We'll just miss helix testing our bits against internal-only runtime changes, but we'll still get coverage from the non-helix jobs.

Reply:
We get very little coverage from non-Helix jobs because tests that run in Helix don't run anywhere else in our pipelines. And, the switch can't be forgotten because PRs will fail unless it's set correctly.

Reply:
> the switch can't be forgotten because PRs will fail unless it's set correctly.

This is the part that concerns me, I doubt devs or CQBs would know where to look to fix errors caused by this, even if we communicated it in advance. We already are inconsistent enough about following process that adding more of it (and a new class of it) worries me. I guess there's no way to always get test coverage pre-https://github.com/dotnet/arcade/issues/8336, without manually updating _something_ based on the state of where we're getting our dependencies, but I'd prefer doing that in code (some property in the .yml file itself) rather than in the AzDO UI. I'd prefer that or the ContinueOnError approach to hosting the switch in a separate location like the UI, what do you prefer?

Reply:
🆗 that's a good compromise: Put the variable in the YAML files w/ a nice big comment.

Reply:
The other option would be a property w/ a nice big comment at the top of this file plus changing the YAML to always pass _both_ tokens (one base64, one not) in.

Reply:
I think https://github.com/dotnet/aspnetcore/pull/40073/commits/fd80be8577a072a935e835c89229376c04cf8538 is all we need given the other conversation above